From db57f3c6d11864f4227bb6734a0d509695a32a7d Mon Sep 17 00:00:00 2001
From: Daniel Mack <zonque@gmail.com>
Date: Thu, 20 Feb 2014 13:29:54 +0100
Subject: [PATCH 1/2] core/unit: fix unit_add_target_dependencies() for units
 with no dependencies

For units without any dependencies, r needs to be initialized to 0.
Otherwise, the return value of unit_add_target_dependencies() is
unspecified.
---
 src/core/unit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/unit.c b/src/core/unit.c
index 21ad434..016b1e3 100644
--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -959,7 +959,7 @@ static int unit_add_target_dependencies(Unit *u) {
         Unit *target;
         Iterator i;
         unsigned k;
-        int r;
+        int r = 0;
 
         assert(u);
 
-- 
1.9.0

